v2/features/connection/WalletToggle.tsx

```diff
@@ -50,16 +51,17 @@ const WalletToggleOptions: FC = () => {
 };
 
 const WalletToggle: FC = () => {
-  const { account, library } = useWeb3React<Web3Provider>();
-
+  const { account, library, chainId } = useWeb3React<Web3Provider>();
   if (!account) return <ConnectWalletButton />;
 
   return (
     <Popover className="relative">
       {() => (
         <>
           <Popover.Button className="group rounded-xl inline-flex items-center text-sm text-white font-bold hover:bg-black-light transition duration-300 ease-in-out focus:outline-none px-4 py-2">
-            <span className="block mr-2">{shortenAddress(account)}</span>
+            <span className="block mr-2">
+              {useENS(account, library, chainId) || shortenAddress(account)}
```

Comment:
@amatureApe just a quick note, while this works (?), it's a good practice to call hooks at the top of the function. This will lead to problems in more complex scenarios. See [rules of hooks](https://reactjs.org/docs/hooks-rules.html).

Reply:
Ok, will do. Do you want me to change it here or leave it as is?

Reply:
Can you please do it in a follow up PR? In theory this shouldn't work or at least throw a warning 😅

Reply:
Ok I've pushed it. Hopefully it is set up how you wanted it.